core/src/main/java/org/bitcoinj/evolution/ProviderRegisterTx.java

```diff
@@ -81,7 +81,7 @@ protected void parse() throws ProtocolException {
         mode = readUint16();
         collateralOutpoint = new TransactionOutPoint(params, payload, cursor);
         cursor += collateralOutpoint.getMessageSize();
-        address = new MasternodeAddress(params, payload, cursor, NetworkParameters.ProtocolVersion.CURRENT.getBitcoinProtocolVersion());
+        address = new MasternodeAddress(params, payload, cursor, params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.CURRENT));
```

Comment:
I was using the wrong getter for the protocol version in many places.  

When we want to get the current protocol version, we should use:
```
params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.CURRENT)
```
because `getProtocolVersionNum` can be overridden by a network.